graphql/schema.graphql

```diff
@@ -91,6 +92,27 @@ input DisplayTask {
   ExecTasks: [String!]!
 }
 
+input SubscriptionInput {
+  resource_type: String
+  trigger: String
+  selectors: [SelectorInput!]!
+  regex_selectors: [SelectorInput!]!
+  subscriber: SubscriberInput!
+  owner_type: String
+  owner: String
+  trigger_data: StringMap!
+}
+
+input SelectorInput {
+  type: String!
```

Comment:
do you think these would benefit from being an enum? i imagine there is a limited set of values this can be--one of them being slack.

Reply:
Adding enums for input fields doesn't work well when we the gql type is built from an existing go struct. I tried assigning an enum to the resource_type in the SubscriptionInput gql type. gqlgen is unsure how to convert the enum type to *string when unmarshaling the SubscriptionInput gql input to APISubscription